### PR TITLE
chore: Add permissions for interruption-handling tests

### PR DESCRIPTION
**Pull request by jonathan-innis**
Issue #, if available:

Description of changes:

- Adds the following permissions to enable Karpenter E2E testing for interruption:

- `sqs:*`
- `fis:*`

A managed policy for Fault injection service `service-role/AWSFaultInjectionSimulatorEC2Access`
Add `sts:AssumeRole` for the FIS service

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by hakuna-matatah — APPROVED** [review]
lgtm

File: infrastructure/lib/kit-infrastructure.ts [hunk 1]
```diff
@@ -139,11 +139,26 @@ export class KITInfrastructure extends Stack {
...
+    // Add FIS permissions for EC2 interruptions
+    sa.role.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSFaultInjectionSimulatorEC2Access'))
+    sa.role.grantAssumeRole(new iam.ServicePrincipal("fis.amazonaws.com", 
+    {
+      conditions: {
+        StringEquals: {
+          "aws:SourceAccount": Stack.of(this).account,
+        },
+        ArnLike: {
+          "aws:SourceArn": `arn:${Stack.of(this).partition}:fis:${Stack.of(this).region}:${Stack.of(this).account}:experiment/*`
```
Comment (hakuna-matatah) [inline infrastructure/lib/kit-infrastructure.ts:158]: Why does it say `experiment` ? Is this not GA ?

Comment (jonathan-innis) [reply infrastructure/lib/kit-infrastructure.ts]: experiments is an FIS resource owned by the FIS service. It has nothing to do with the lifecycle of the resource.

**Review by bwagner5 — APPROVED** [review]
LGTM

**Pull request merged**